### PR TITLE
feat: paid tunnels v1

### PR DESCRIPTION
**Pull request by charlottea98**
Tested going from paid to free tunnel, and vice versa. Works smoothly for me at least 🤞 

Some extra logging added that shows up when running `RUST_LOG=info cargo run -- start`. There is more that can be done with logging but will leave that for the linear card DO-1495 "linkup: logger".

Part of DO-1495
Resolves DO-1496

### CODE REVIEW

File: linkup-cli/src/paid_tunnel.rs [hunk 1]
```diff
@@ -138,13 +139,21 @@ impl PaidTunnelManager for RealPaidTunnelManager {
...
         let (client, headers) = prepare_client_and_headers(&RealSystem)?;
-        let query_url = format!("{}?name=tunnel-{}", url, tunnel_name);
+        let query_url = format!("{}?name={}", url, tunnel_name);
 
         let parsed: GetTunnelApiResponse = send_request(&client, &query_url, headers, None, "GET")?;
         if parsed.result.is_empty() {
             Ok(None)
         } else {
-            Ok(Some(parsed.result[0].id.clone()))
+            // Check if there exists a tunnel with this name that hasn't been deleted
```
Comment (charlottea98) [inline linkup-cli/src/paid_tunnel.rs:148]: Deleting a tunnel doesn't delete it, it just sets "deleted_at". Instead, it needs to find the tunnel name and where "deleted_at" is null

File: linkup-cli/src/services/tunnel.rs [hunk 2]
```diff
@@ -61,6 +54,13 @@ impl TunnelManager for RealTunnelManager {
             }
         }
     }
+    fn is_tunnel_running(&self) -> Result<(), CheckErr> {
+        if !linkup_file_path(LINKUP_CLOUDFLARED_PID).exists() {
```
Comment (jonaslindstr) [inline linkup-cli/src/services/tunnel.rs:58]: Condition can be reversed

**Review by jonaslindstr — APPROVED** [review]

File: clear-unused-dns.ts [hunk 3]
```diff
@@ -56,18 +96,22 @@ async function deleteTXTRecords(
...
+const account_id = process.env.LINKUP_CLOUDFLARE_ACCOUNT_ID;
+const mentimeter_dev_zone_id = process.env.LINKUP_CLOUDFLARE_ZONE_ID;
 
-if (!apikey || !email) {
-  console.error("Missing Cloudflare API key or email");
+if (!api_token || !account_id || !mentimeter_dev_zone_id) {
+  console.error("Missing Cloudflare API Token, Account ID or Zone ID");
   console.error(
-    "Please set CLOUDFLARE_API_KEY and CLOUDFLARE_EMAIL environment variables"
+    "Run `menti localsecrets` to set the required environment variables"
```
Comment (ostenbom) [inline clear-unused-dns.ts:106]: This doesn't make sense in an open-source repo - but maybe we leave it / it's ok?

**Review by ostenbom — APPROVED** [review]
You can bump linkup-cli version in it's cargo.toml version too!

**Pull request merged**